framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -2713,9 +2733,11 @@ public Void visitNewArray(NewArrayTree tree, Void p) {
     /**
      * If the lint option "cast:redundant" is set, this method issues a warning if the cast is
      * redundant.
+     *
+     * @param typeCastTree the TypeCastTree to check
      */
     protected void checkTypecastRedundancy(TypeCastTree typeCastTree) {
-        if (!checker.getLintOption("cast:redundant", false)) {
+        if (!(lintCastRedundantEnabled || lintCastEnabled)) {
```

Comment:
Fold these into the computation in the initializer. No need to re-compute this every time, as the values don't change.